# ***ydata-profiling checkpoint 2*** 

Objective
In this checkpoint, we are going to work on 'The Tunisian flight company: Tunisair' dataset.
Dataset description : This dataset include historical data of flight delays. the objective is to investigate the flight delays factors.
###

## ***INSTRUCTIONS***
- Part 1: Data Exploration with Pandas
- Load the provided dataset into a pandas dataframe
- Use pandas to explore the dataset. For example, you might start by using the head() and info() methods to get an overview of the data.
- Look for missing values in the dataset. You can use the isnull() method to identify missing values.
- Use pandas to calculate some summary statistics for the dataset. For example, you might use the describe() method to get summary statistics for the numerical columns in the dataset.
- Part 2: Data Exploration with ydata-Profiling
- Use ydata-profiling to generate a report of the provided dataset.
- Look for missing values in the dataset. You can use the report generated by ydata-profiling to identify missing values.
- Look for correlations between different columns in the dataset. You can use the report generated by ydata-profiling to identify correlations between different columns.
- Identify any outliers or unusual values in the dataset. You can use the report generated by ydata-profiling to identify any outliers or unusual values.
***Summary***
At the end of this exercise, write a summary of your findings. Did you find any interesting patterns or correlations in the data? Were there any issues or challenges you encountered while exploring the dataset? Use your summary to reflect on your experience using pandas and ydata-profiling to explore and understand a new dataset.

In [1]:
import pandas as pd
import seaborn as sns
from ydata_profiling import ProfileReport

In [2]:
df = pd.read_csv(r"C:\Users\DELL\Downloads\Tunisair_flights_dataset.csv")

In [4]:
# first 20 values 
data_profiling = df.head(20)
data_profiling

Filght_date Flight_ID Departure point Arrival point  \
0   2016-01-03  TU 0712              CMN           TUN   
1   2016-01-13  TU 0757              MXP           TUN   
2   2016-01-16  TU 0214              TUN           IST   
3   2016-01-17  TU 0480              DJE           NTE   
4   2016-01-17  TU 0338              TUN           ALG   
5   2016-01-17  TU 0283              TLS           TUN   
6   2016-01-18  TU 0514              TUN           BCN   
7   2016-01-18  TU 0716              TUN           ORY   
8   2016-01-18  TU 0752              TUN           FCO   
9   2016-01-18  TU 0996              TUN           NCE   
10  2016-01-18  TU 0930              TUN           MRS   
11  2016-01-20  TU 0724              TUN           ORY   
12  2016-01-20  TU 0913              TUN           MED   
13  2016-01-21  TU 0217              IST           TUN   
14  2016-01-22  TU 0744              TUN           FRA   
15  2016-01-22  TU 0706              CMN           TUN   
16  2016-01-22  TU 0723              ORY           TUN   
17  2016-01-23  TU 0726              MIR           BRU   
18  2016-01-22  TU 0913              TUN           MED   
19  2016-01-23  TU 9038              TUN           DJE   

   Scheduled_departure_time Scheduled_arrival_time STATUS Aircraft_code  \
0       2016-01-03 10:30:00    2016-01-03 12.55.00    ATA     TU 32AIMN   
1       2016-01-13 15:05:00    2016-01-13 16.55.00    ATA     TU 31BIMO   
2       2016-01-16 04:10:00    2016-01-16 06.45.00    ATA     TU 32AIMN   
3       2016-01-17 14:10:00    2016-01-17 17.00.00    ATA     TU 736IOK   
4       2016-01-17 14:30:00    2016-01-17 15.50.00    ATA     TU 320IMU   
5       2016-01-17 16:20:00    2016-01-17 18.15.00    ATA     TU 736IOP   
6       2016-01-18 07:15:00    2016-01-18 09.00.00    ATA     TU 32AIMH   
7       2016-01-18 07:35:00    2016-01-18 09.55.00    ATA     TU 32AIMI   
8       2016-01-18 07:40:00    2016-01-18 09.00.00    ATA     TU 32AIMC   
9       2016-01-18 07:45:00    2016-01-18 09.15.00    ATA     TU 31AIMK   
10      2016-01-18 07:55:00    2016-01-18 09.30.00    ATA     TU 32AIMF   
11      2016-01-20 18:30:00    2016-01-20 20.50.00    ATA     TU 32AIMP   
12      2016-01-20 19:40:00    2016-01-21 00.00.00    ATA     TU 320IMR   
13      2016-01-21 20:10:00    2016-01-21 23.00.00    ATA     TU 320IMU   
14      2016-01-22 07:05:00    2016-01-22 09.40.00    ATA     TU 736IOP   
15      2016-01-22 17:45:00    2016-01-22 20.10.00    ATA     TU 320IMR   
16      2016-01-22 18:30:00    2016-01-22 21.00.00    ATA     TU 332IFN   
17      2016-01-23 05:30:00    2016-01-23 08.15.00    ATA     TU 31AIMJ   
18      2016-01-22 19:40:00    2016-01-23 00.00.00    ATA     TU 320IMT   
19      2016-01-23 08:50:00    2016-01-23 09.45.00    ATA     TU 736IOL   

    Arrival delay  
0           260.0  
1            20.0  
2             0.0  
3             0.0  
4            22.0  
5            53.0  
6            10.0  
7            15.0  
8            16.0  
9            21.0  
10           12.0  
11           18.0  
12            0.0  
13            0.0  
14           18.0  
15            0.0  
16           37.0  
17          149.0  
18           30.0  
19           19.0

In [5]:
data_profiling.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 20 entries, 0 to 19
Data columns (total 9 columns):
 #   Column                    Non-Null Count  Dtype  
---  ------                    --------------  -----  
 0   Filght_date               20 non-null     object 
 1   Flight_ID                 20 non-null     object 
 2   Departure point           20 non-null     object 
 3   Arrival point             20 non-null     object 
 4   Scheduled_departure_time  20 non-null     object 
 5   Scheduled_arrival_time    20 non-null     object 
 6   STATUS                    20 non-null     object 
 7   Aircraft_code             20 non-null     object 
 8   Arrival delay             20 non-null     float64
dtypes: float64(1), object(8)
memory usage: 1.5+ KB


In [6]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 107833 entries, 0 to 107832
Data columns (total 9 columns):
 #   Column                    Non-Null Count   Dtype  
---  ------                    --------------   -----  
 0   Filght_date               107833 non-null  object 
 1   Flight_ID                 107833 non-null  object 
 2   Departure point           107833 non-null  object 
 3   Arrival point             107833 non-null  object 
 4   Scheduled_departure_time  107833 non-null  object 
 5   Scheduled_arrival_time    107833 non-null  object 
 6   STATUS                    107833 non-null  object 
 7   Aircraft_code             107833 non-null  object 
 8   Arrival delay             107833 non-null  float64
dtypes: float64(1), object(8)
memory usage: 7.4+ MB


In [7]:
# get missing values
missing_values = print(df.isnull().sum())
missing_values


Filght_date                 0
Flight_ID                   0
Departure point             0
Arrival point               0
Scheduled_departure_time    0
Scheduled_arrival_time      0
STATUS                      0
Aircraft_code               0
Arrival delay               0
dtype: int64


In [8]:
df.describe()

Arrival delay
count  107833.000000
mean       48.733013
std       117.135562
min         0.000000
25%         0.000000
50%        14.000000
75%        43.000000
max      3451.000000

In [9]:
profile = ProfileReport(df, title='Tunisair flight dataset Profile', explorative=True)
profile.to_file("Tunisair_flight_dataset_profile_report.html")

Summarize dataset:   0%|          | 0/5 [00:00<?, ?it/s]

Generate report structure:   0%|          | 0/1 [00:00<?, ?it/s]

Render HTML:   0%|          | 0/1 [00:00<?, ?it/s]

Export report to file:   0%|          | 0/1 [00:00<?, ?it/s]

# Look for missing values in the dataset. You can use the report generated by ydata-profiling to identify missing values
***THERE ARE NO MISSING VALUES***

# Look for correlations between different columns in the dataset. You can use the report generated by ydata-profiling to identify correlations between different columns.
***according to the heat map in the ydata_profile_report,as Arrival delay increases,the status variable tends to be lower and vice versa.***

**SUMMARY**

***The YData profile report details a dataset containing 107,833 observations and 9 variables, with no missing or duplicate values, and a total memory size of 55.3 MiB. The dataset comprises 2 DateTime, 5 Text, 1 Categorical, and 1 Numeric variable. Key variables include Flight_date, Flight_ID, Departure point, Arrival point, Scheduled_departure_time, Scheduled_arrival_time, STATUS, Aircraft_code, and Arrival delay. Flight_date ranges from January 1, 2016, to December 31, 2018. STATUS has 5 categories: ATA (93,679 instances), SCH (13,242), DEP (467), RTR (294), and DEL (151). The mean Arrival delay is 48.7 minutes, with 35.4% of flights having no delay. Sample data includes details such as flight dates, IDs, departure and arrival points, scheduled times, statuses, aircraft codes, and delays.***